src/components/common/toast/toast.tsx

```diff
@@ -0,0 +1,73 @@
+import { Toast } from "@lifesg/react-design-system/toast";
+import { useEffect, useRef } from "react";
+import { EToastTypes, IToast, useDisplay } from "src/context-providers";
+import { ToastWrapper } from "./toast.styles";
+
+interface IProps {
+    toast?: IToast;
+    toastFunction?: () => void;
+    index?: number;
+}
+
+export const DisplayToast = ({ toast, toastFunction, index }: IProps) => {
+    // =========================================================================
+    // CONST, STATE, REF
+    // =========================================================================
+    const { dismissToast } = useDisplay();
+    const timeoutRef = useRef(null);
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    function startDismissTimer(id: string) {
+        return setTimeout(() => {
+            dismissToast(id);
+        }, 4500);
+    }
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        if (toast.id !== undefined) {
```

Comment:
under what scenario will the id be undefined?

Reply:
changes made in the last push